api/base/parsers.py

```diff
@@ -8,3 +9,25 @@ class JSONAPIParser(JSONParser):
     """
     media_type = 'application/vnd.api+json'
     renderer_class = JSONAPIRenderer
+
+    def parse(self, stream, media_type=None, parser_context=None):
```

Comment:
Discussed in person: This processing should also happen for requests that have the `application/json...` media type.